hypothesis-python/src/hypothesis/internal/compat.py

```diff
@@ -125,14 +125,24 @@ def get_type_hints(thing):
 else:
 
     def get_type_hints(thing):
-        if inspect.isclass(thing) and not hasattr(thing, "__signature__"):
-            if is_typed_named_tuple(thing):
-                # Special handling for typing.NamedTuple
-                return thing._field_types  # type: ignore
-            thing = thing.__init__  # type: ignore
+        """Like the typing version, but tries harder and never errors.
+
+        Tries harder: if the thing to inspect is a class and raises an error or we
+        find no hints, this function will then inspect the __init__ method.  This
+        often helps with user-defined classes on older versions of Python.
```

Comment:
sorry for the nit, but this actually was hard to parse at first

```suggestion
        Tries harder: if the thing to inspect is a class, but it raises an error or we
        find no hints, then this function will then inspect the __init__ method.  This
        often helps with user-defined classes on older versions of Python.
```